tests/test_predicates.py

```diff
@@ -144,6 +105,6 @@ def test_lift_matrix(self):
         self.assertTrue(equal(target, lift_matrix(base, list_test)))
         self.assertTrue(equal(target, lift_matrix(base, vector_test)))
         self.assertTrue(equal(target, lift_matrix(base, matrix_test)))
-    '''
```

Comment:
well that would probably go into a different branch and PR. If you have a better implementation we should probably use it. But in a different branch, this PR is already beefy enough lol